File: code/js/src/services/media/siren/SirenEntity.ts
import { Link } from './Link';
import { Action } from './Action';
import { SubEntity } from './SubEntity';
import { Entity } from './Entity';

export class SirenEntity<T> implements Entity<T> {
  constructor(
    public clazz?: string[],
    public properties?: T,
    public links?: Link[],
    public actions?: Action[],
    public entities?: SubEntity[]
  ) {}
}

File: code/jvm/src/test/kotlin/pt/isel/daw/gomoku/http/hypermedia/SirenEntity.kt
package pt.isel.daw.gomoku.http.hypermedia

import com.fasterxml.jackson.annotation.JsonProperty

data class SirenEntity<T>(
    @get:JsonProperty("class")
    override val clazz: List<String>? = null,
    override val properties: T? = null,
    override val links: List<Link>? = null,
    override val actions: List<Action>? = null,
    override val entities: List<SubEntity>? = null
) : Entity<T>
